core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedRegionPlanGenerator.scala

```diff
@@ -132,17 +136,31 @@ class CostBasedRegionPlanGenerator(
     * @return A region DAG.
     */
   private def createRegionDAG(): DirectedAcyclicGraph[Region, RegionLink] = {
-    val searchResult = if (AmberConfig.useTopDownSearch) {
-      topDownSearch(globalSearch = AmberConfig.useGlobalSearch)
-    } else {
-      bottomUpSearch(globalSearch = AmberConfig.useGlobalSearch)
+    val searchResultFuture: Future[SearchResult] = Future {
+      if (AmberConfig.useTopDownSearch)
+        topDownSearch(globalSearch = AmberConfig.useGlobalSearch)
+      else
+        bottomUpSearch(globalSearch = AmberConfig.useGlobalSearch)
+    }
+    val searchResult = Try(
+      Await.result(searchResultFuture, AmberConfig.searchTimeoutMilliseconds.milliseconds)
+    ) match {
+      case Failure(exception) =>
```

Comment:
can we do a check on the type of exception and only handle time out explicitly? for any other possible exceptions, we may want to fail the search right away.

Reply:
added